python/cugraph/cugraph/centrality/plc_betweenness_centrality.py

```diff
@@ -0,0 +1,189 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pylibcugraph import (
+    betweenness_centrality as pylibcugraph_betweenness_centrality,
+    ResourceHandle,
+)
+
+from cugraph.utilities import (
+    ensure_cugraph_obj_for_nx,
+    df_score_to_dictionary,
+)
+import cudf
+import warnings
+import numpy as np
+
+
+def plc_betweenness_centrality(
+    G,
+    k=None,
+    normalized=True,
+    weight=None,
+    endpoints=False,
+    seed=None,
+    result_dtype=np.float64,
+):
+    """
+    Compute the betweenness centrality for all vertices of the graph G.
+    Betweenness centrality is a measure of the number of shortest paths that
+    pass through a vertex.  A vertex with a high betweenness centrality score
+    has more paths passing through it and is therefore believed to be more
+    important.
+
+    To improve performance. rather than doing an all-pair shortest path,
+    a sample of k starting vertices can be used.
+
+    CuGraph does not currently support the 'endpoints' and 'weight' parameters
+    as seen in the corresponding networkX call.
+
+    Parameters
+    ----------
+    G : cuGraph.Graph or networkx.Graph
+        The graph can be either directed (Graph(directed=True)) or undirected.
+        Weights in the graph are ignored, the current implementation uses a parallel
+        variation of the Brandes Algorithm (2001) to compute exact or approximate
+        betweenness. If weights are provided in the edgelist, they will not be
+        used.
+
+    k : int or list or None, optional (default=None)
+        If k is not None, use k node samples to estimate betweenness.  Higher
+        values give better approximation.  If k is a list, use the content
+        of the list for estimation: the list should contain vertex
+        identifiers. If k is None (the default), all the vertices are used
+        to estimate betweenness.  Vertices obtained through sampling or
+        defined as a list will be used as sources for traversals inside the
+        algorithm.
+
+    normalized : bool, optional (default=True)
+        If true, the betweenness values are normalized by
+        __2 / ((n - 1) * (n - 2))__ for undirected Graphs, and
+        __1 / ((n - 1) * (n - 2))__ for directed Graphs
+        where n is the number of nodes in G.
+        Normalization will ensure that values are in [0, 1],
+        this normalization scales for the highest possible value where one
+        node is crossed by every single shortest path.
+
+    weight : cudf.DataFrame, optional (default=None)
+        Specifies the weights to be used for each edge.
+        Should contain a mapping between
+        edges and weights.
+
+        (Not Supported): But if weights are provided at the Graph creation,
+        they will be used
+
+    endpoints : bool, optional (default=False)
+        If true, include the endpoints in the shortest path counts.
+
+    seed : optional (default=None)
```

Comment:
I think @rlratzel and I agreed that this parameter should be called "random_state" everywhere.

Reply:
Right.